File: SEP490_Ezstay/AccountAPI/Service/PhoneOtpClient.cs
using AccountAPI.Service.Interfaces;

namespace AccountAPI.Service
{
    public class PhoneOtpClient : IPhoneOtpClient
    {
        private readonly HttpClient _http;

        public PhoneOtpClient(HttpClient http)
        {
            _http = http;
        }

        public async Task<bool> SendOtpAsync(string phone)
        {
            var res = await _http.PostAsJsonAsync("/api/Auth/send-phone-otp", new { phone });
            return res.IsSuccessStatusCode;
        }

        public async Task<bool> VerifyOtpAsync(string phone, string otp)
        {
            var res = await _http.PostAsJsonAsync("/api/Auth/verify-phone-otp", new { phone, otp });
            return res.IsSuccessStatusCode;
        }
    }
}

File: SEP490_Ezstay/AccountAPI/AccountAPI/Service/PhoneOtpClient.java
package AccountAPI.Service;

import AccountAPI.Service.Interfaces.IPhoneOtpClient;

public class PhoneOtpClient implements IPhoneOtpClient {
	private HttpClient _http;

	public PhoneOtpClient(HttpClient aHttp) {
		throw new UnsupportedOperationException();
	}

	public async_Task<boolean> SendOtpAsync(String aPhone) {
		throw new UnsupportedOperationException();
	}

	public async_Task<boolean> VerifyOtpAsync(String aPhone, String aOtp) {
		throw new UnsupportedOperationException();
	}
}